### PR TITLE
🚨🚨🐛Source Google Ads: upgrade API version to v15

### LINKED ISSUE #33029: Source Google Ads must urgently upgrade from API v13
### Topic

Google is sunsetting v13 of the Google Ads API in January 2024, after which all requests to that API version will fail. That is the version used by our Google Ads connector.

### Relevant information

- Google Ads API docs: https://developers.google.com/google-ads/api/docs/sunset-dates#timetable
- [user report in Airbyte slack](https://airbytehq-team.slack.com/archives/C027KKE4BCZ/p1701417882881009)

As the google API docs say,
> you don't have to upgrade in strict sequential order—for instance, you can upgrade from version (N) directly to version (N+2).

It seems to me we should upgrade straight from v13 to v15 if it wouldn't take significantly longer.

### PR DESCRIPTION
**Pull request by askarpets**
## What
Resolve https://github.com/airbytehq/airbyte/issues/33029

- Upgrade Google Ads API version to v15
- Rename/delete the following:

| Stream                     | Current field name                                                         | New field name                                                           |
|----------------------------|----------------------------------------------------------------------------|--------------------------------------------------------------------------|
| ad_listing_group_criterion | ad_group_criterion.listing_group.case_value.product_bidding_category.id    | ad_group_criterion.listing_group.case_value.product_category.category_id |
| ad_listing_group_criterion | ad_group_criterion.listing_group.case_value.product_bidding_category.level | ad_group_criterion.listing_group.case_value.product_category.level       |
| shopping_performance_view  | segments.product_bidding_category_level1                                   | segments.product_category_level1                                         |
| shopping_performance_view  | segments.product_bidding_category_level2                                   | segments.product_category_level2                                         |
| shopping_performance_view  | segments.product_bidding_category_level3                                   | segments.product_category_level3                                         |
| shopping_performance_view  | segments.product_bidding_category_level4                                   | segments.product_category_level4                                         |
| shopping_performance_view  | segments.product_bidding_category_level5                                   | segments.product_category_level5                                         |
| campaign                   | campaign.shopping_setting.sales_country                                    | This field has been deleted                                              |

See docs for more info:

- [ad_listing_group_criterion](https://developers.google.com/google-ads/api/fields/v15/ad_group_criterion)
- [shopping_performance_view](https://developers.google.com/google-ads/api/fields/v15/shopping_performance_view)
- [campaign](https://developers.google.com/google-ads/api/fields/v15/campaign)

## 🚨 User Impact 🚨
This is a breaking change since several fields were renamed/deleted

## Pre-merge Actions

<details><summary><strong>Updating a connector</strong></summary>

### Community member or Airbyter

- Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- Unit & integration tests added

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- Create a non-forked branch based on this PR and test the below items on it
- Build is successful
- If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).

</details>

### CODE REVIEW

**Review by darynaishchenko — APPROVED** [review]

**Review by tolik0 — APPROVED** [review]

**Review by artem1205 — APPROVED** [review]

File: airbyte-integrations/connectors/source-google-ads/metadata.yaml [hunk 1]
```diff
@@ -39,6 +39,12 @@ data:
           API. Users should refresh the source schema and reset affected streams after
           upgrading to ensure uninterrupted syncs.
         upgradeDeadline: "2023-11-30"
+      3.0.0:
+        message:
+          This release upgrades the Google Ads API to version 15 which causes changes in several schemas.
+          Users should refresh the source schema and reset affected streams after
+          upgrading to ensure uninterrupted syncs.
+        upgradeDeadline: "2023-12-25"
```
Comment (katmarkham) [inline airbyte-integrations/connectors/source-google-ads/metadata.yaml:47]: Google is deprecating v13 of the Google Ads API in January. This release upgrades the Google Ads API to the latest version (v15), which causes changes in several schemas. Users should refresh the source schema and reset affected streams after upgrading to ensure uninterrupted syncs.

Comment (katmarkham) [inline airbyte-integrations/connectors/source-google-ads/metadata.yaml:47]: "2023-12-29"

File: docs/integrations/sources/google-ads-migrations.md [hunk 2]
```diff
@@ -1,5 +1,24 @@
 # Google Ads Migration Guide
 
+## Upgrading to 3.0.0
+
+This release upgrades the Google Ads API to version 15 which causes the following changes in the schemas:
```
Comment (katmarkham) [inline docs/integrations/sources/google-ads-migrations.md:5]: This release upgrades the Google Ads API from Version 13 to Version 15 which causes the following changes in the schemas:

File: docs/integrations/sources/google-ads-migrations.md [hunk 3]
```diff
@@ -1,5 +1,24 @@
...
+| shopping_performance_view  | segments.product_bidding_category_level2                                   | segments.product_category_level2                                         |
+| shopping_performance_view  | segments.product_bidding_category_level3                                   | segments.product_category_level3                                         |
+| shopping_performance_view  | segments.product_bidding_category_level4                                   | segments.product_category_level4                                         |
+| shopping_performance_view  | segments.product_bidding_category_level5                                   | segments.product_category_level5                                         |
+| campaign                   | campaign.shopping_setting.sales_country                                    | This field has been deleted                                              |
+
+Users should:
+- Refresh the source schema
+- And reset affected streams after upgrading to ensure uninterrupted syncs.
```
Comment (katmarkham) [inline docs/integrations/sources/google-ads-migrations.md:20]: Reset affected streams after upgrading to ensure uninterrupted syncs.

**Review by katmarkham — CHANGES_REQUESTED** [review]

File: airbyte-integrations/connectors/source-google-ads/metadata.yaml (hunk 1 above)
Comment (askarpets) [reply airbyte-integrations/connectors/source-google-ads/metadata.yaml]: Updated

Comment (askarpets) [reply airbyte-integrations/connectors/source-google-ads/metadata.yaml]: Updated

File: docs/integrations/sources/google-ads-migrations.md (hunk 2 above)
Comment (askarpets) [reply docs/integrations/sources/google-ads-migrations.md]: Updated

File: docs/integrations/sources/google-ads-migrations.md (hunk 3 above)
Comment (askarpets) [reply docs/integrations/sources/google-ads-migrations.md]: Updated

**Review by katmarkham — APPROVED** [review]

**Pull request merged**